python/cudaq/handlers/photonics_kernel.py

```diff
@@ -0,0 +1,213 @@
+class PyQudit:
+    """
+    A data structure to represent `qudit` which models a general d-level 
+    quantum system.
+
+    Args:
+        level (int): An integer representing the level of quantum system to 
+            which this qudit belongs to.
+        id (int): An integer representing the unique identifier for this qudit
+            in the system.
+    """
+    level: int
+    id: int
+
+    def __del__(self):
+        cudaq_runtime.photonics.release_qudit(self.level, self.id)
+
+
+def _is_qudit_type(q: any) -> bool:
+    """
+    Utility function to check whether the input argument is instance of 
+    `PyQudit` class.
+
+    Returns:
+        bool: `True` if input argument is instance or a list of `PyQudit` 
+               class, else `False`
+    """
+    if isinstance(q, PyQudit):
+        return True
+    if isinstance(q, List):
+        return all(isinstance(i, PyQudit) for i in q)
+    return False
+
+
+def _check_args(q: any):
+    """
+    Utility function to verify the arguments to a photonic quantum operation.
+
+    Args:
+        q: A single instance or a list of objects of `PyQudit` class.
+
+    Raises:
+        RuntimeError: If the qudit level is not set.
+        Exception: If input argument is not instance of `PyQudit` class.
+    """
+    if globalQuditLevel is None:
+        raise RuntimeError(
+            "Qudit level not set. Define a qudit (`qudit(level=N)`) or list of qudits."
+        )
+
+    if not _is_qudit_type(q):
+        raise Exception(
+            "Invalid quantum type. Use qudit (`qudit(level=N)`) or a list of qudits."
+        )
+
+
+def qudit(level: int) -> PyQudit:
+    """
+    API to define a d-level qudit in a `PhotonicsHandler`. All the qudits within
+    a kernel are of the same level.
+
+    Args:
+        level (int): An integer representing the level of quantum system.
+    
+    Returns:
+        An instance of `PyQudit`.
+    
+    Raises:
+        RuntimeError: If a qudit of level different than one already defined 
+            in the kernel is requested.
+    """
+    global globalQuditLevel
+
+    if globalQuditLevel is None:
+        globalQuditLevel = level
+    elif level != globalQuditLevel:
+        raise RuntimeError("The qudits must be of same level within a kernel.")
+
+    id = cudaq_runtime.photonics.allocate_qudit(globalQuditLevel)
+    return PyQudit(globalQuditLevel, id)
+
+
+def plus(qudit: PyQudit):
+    """
+    Apply plus gate on the input qudit. 
+    U|0> -> |1>, U|1> -> |2>, ..., and U|d> -> |0>
+
+    Args: 
+        qudit: An instance of `PyQudit` class.
+    
+    Raises:
+        RuntimeError: If the qudit level is not set.
+        Exception: If input argument is not instance of `PyQudit` class.
+    """
+    _check_args(qudit)
+    cudaq_runtime.photonics.apply_operation('plusGate', [],
+                                            [[qudit.level, qudit.id]])
+
+
+def phase_shift(qudit: PyQudit, phi: float):
+    """
+    Apply phase shift gate.
+    TBD
```

Comment:
Is resolving these `TBD`s planned for the remainder of #1937? (Same for lines 141-145)

Reply:
Yes, external contribution expected on the documentation, including gate descriptions.